kazoo/handlers/gevent.py

```diff
@@ -69,14 +69,17 @@ def greenlet_worker():
             while True:
                 try:
                     func = queue.get()
-                    if func is _STOP:
-                        break
-                    func()
+                    try:
+                        if func is _STOP:
+                            break
+                        func()
+                    except Exception as exc:
+                        log.warning("Exception in worker greenlet")
+                        log.exception(exc)
+                    finally:
+                        del func  # release before possible idle
```

Comment:
> Given the attention to detail in your PR's, I suspect I'm missing something... Is there some reason that it's better to set it to None?

I had set the `func` to `None` for preventing from `NameError` if an exception was raised earlier than `func` was assigned. But it is confusing and `del func` is more explicit that it, I agree with you.

I use the nested `try-except` like the implementation of `SequentialThreadingHandler` now. The `NameError` will never occur here.